csharp/RowOriented/ParquetFile.cs

```diff
@@ -20,35 +20,49 @@ public static class ParquetFile
         /// <summary>
         /// Create a row-oriented reader from a file.
         /// </summary>
-        public static ParquetRowReader<TTuple> CreateRowReader<TTuple>(string path)
+        public static ParquetRowReader<TTuple> CreateRowReader<TTuple>(
+            string path,
+            LogicalTypeFactory? logicalTypeFactory = null,
+            LogicalReadConverterFactory? logicalReadConverterFactory = null)
```

Comment:
I haven't taken a good look at all of your code changes but have one comment here. We try to avoid breaking binary compatibility so that users can use a new version of ParquetSharp in an application built against an older version. This means that rather than adding new optional parameters to existing public methods, you'll need to keep the existing methods as they are and add new overloads that take additional parameters.  There's a good article describing breaking changes in .NET libraries at https://learn.microsoft.com/en-us/dotnet/standard/library-guidance/breaking-changes

Reply:
Makes sense, thank you!

I'll work on rewriting the code as new methods instead. Would you say it's necessary to add an extra overload for each existing method `ParquetFile.CreateRowReader()` and `ParquetFile.CreateRowWriter()`, along with the `ParquetRowReader/Writer` constructors?

I am currently setting the relevant factories in the common constructors for `ParquetRowReader` and `ParquetRowWriter`. Would you say this is the right way to go?

Reply:
I don't think you need to add extra overloads corresponding to each existing method, it might be sufficient to just add one new overload that takes all of the other possible parameters too. Although there might be some cases where you need multiple overloads to accept different input types, eg. reading a stream vs a string for a file path. And yes, setting these in the `ParquetRowReader` and `ParquetRowWriter` looks like a good approach.

Reply:
I'm currently trying to do this with a single overload, however I can't find a way other than checking if one of the parameters is null (say, the `ReaderProperties`) in order to pick a constructor.

I thought perhaps it might be alright to create the additional overloads, because the underlying constructors for `ParquetRowReader`/`ParquetRowWriter` cannot be removed due to the API compatibility/breaking changes issue (suppose if I added a general overload that makes the previous constructors obsolete).

Reply:
Okay yes it's fine to add the extra overloads then if that's the simplest approach. Although the constructors for `ParquetRowReader` and `ParquetRowWriter` are internal only, so you can change their signature if needed. It's only the public factory methods (`CreateRowReader`/`CreateRowWriter`) on `ParquetFile` that need to be kept compatible.